app/bot.py

```diff
@@ -134,23 +134,69 @@ def send_file_info(self, result):
             self.respond("Hmm, I've found a resource here but can't parse it. Moving on...")
 
     def query(self, tokens):
+        
+        #Jas@20/1/16 - If a token has a - in front, it is removed and placed in filter_out
+        def get_filter_terms():
```

Comment:
So there's a couple of stylistic things here - it's much easier to use a list comprehension to do this filtering: 
```python
filter_out = [t.lstrip('-') for t in tokens if t.startswith('-')]
tokens = [t for t in tokens if not t.startswith('-')]
```